src/redux/contacts/reducer.ts

```diff
@@ -99,8 +99,7 @@ const contactReducer = createReducer(initialState, (contacts) => {
       ContactActions.setNewContact.pending,
       (state: ContactsState) => ({
         ...state,
-        loading: true,
-        newContact: {} as Contact
+        loading: true

```

Comment:
Let's change this back to match the comments I mentioned in your action. Then in the create thunk fulfilled case, you can do `{...state, contacts: [...state.contacts, payload]}`. This makes it so if the endpoint is successful you know that it's going to return the object you just added. You can then add that right to the array instead of making another GET to get the new list of contacts

Reply:
I tried to implement this, but it appears that `contacts` is being set to `NULL` with an index as the key maybe? idk if you still have the repo on your machine, but it may be helpful to pull and try adding a new contact to see what happens to the state in the redux dev tool